File: SortTimeComparison/ArrayGenerator.cs
namespace SortTimeComparison
{
    public static class ArrayGenerator
    {
        public static int[] GenerateSortedArray(int n)
        {
            int[] Array = Enumerable.Range(1, n).ToArray();
            return Array;
        }

        public static int[] GenerateReversedSortedArray(int n)
        {
            int[] Array = Enumerable.Range(1, n).Reverse().ToArray();
            return Array;
        }

        public static int[] GenerateRandomArray(int n)
        {
            Random random = new Random();
            int[] Array = Enumerable.Range(1, 100).OrderBy(i => random.Next()).ToArray();
            return Array;
        }
    }
}

File: lab_1_cpp/ArrayGenerator.cpp
//
// Created by Алоев Артем on 17.02.2025.
//

#include "ArrayGenerator.h"

vector<int> ArrayGenerator::GenerateSortedArray(int n) {
    vector<int> array(n);
    for (int i = 0; i < n; ++i) {
        array[i] = i + 1;
    }
    return array;
}

vector<int> ArrayGenerator::GenerateReversedSortedArray(int n) {
    vector<int> array = GenerateSortedArray(n);
    reverse(array.begin(), array.end());
    return array;
}

vector<int> ArrayGenerator::GenerateRandomArray(int n) {
    vector<int> array = GenerateSortedArray(100);
    random_device rd;
    mt19937 g(rd());
    shuffle(array.begin(), array.end(), g);
    return array;
}
